pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+	if err != nil {
+		return err
+	}
+	if launchTemplateId != "" {
+		err = deleteNodeLaunchTemplate(ctx, ec2Client, launchTemplateId)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = deleteRoles(ctx, iamClient, []string{ngRole, *activeCluster.Cluster.RoleArn})
+	if err != nil {
+		return err
+	}
+
+	err = deleteCluster(ctx, eksClient, clusterName)
+	if err != nil {
+		return err
+	}
+
+	return deleteVPC(ctx, ec2Client, *vpcID)
+}
+
+func createCluster(ctx context.Context, eksClient *eks.Client,
+	clusterName, clusterRoleArn, version, cpSgId string, subnetIDs []string,
+	createdByArn string, tags map[string]string) (*types.Cluster, error) {
+	eksCreateInput := &eks.CreateClusterInput{
+		Name:    &clusterName,
+		RoleArn: &clusterRoleArn,
+		Version: aws.String(version),
+
+		AccessConfig: &types.CreateAccessConfigRequest{
+			AuthenticationMode:                      types.AuthenticationModeConfigMap,
+			BootstrapClusterCreatorAdminPermissions: aws.Bool(true),
+		},
+		ResourcesVpcConfig: &types.VpcConfigRequest{
+			EndpointPrivateAccess: aws.Bool(true),
+			EndpointPublicAccess:  aws.Bool(true),
+			SubnetIds:             subnetIDs,
+			SecurityGroupIds:      []string{cpSgId},
+		},
+		KubernetesNetworkConfig: &types.KubernetesNetworkConfigRequest{
+			ServiceIpv4Cidr: aws.String(DefaultKubernetesSvcCIDR),
+		},
+
+		Tags: lo.Assign(
+			map[string]string{TagNameCreateBy: createdByArn},
+			tags,
+		),
+	}
+
+	clusterOutput, err := eksClient.CreateCluster(ctx, eksCreateInput)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create EKS cluster %s", clusterName)
+	}
+	return clusterOutput.Cluster, nil
+}
+
+func buildClusterConfig(clusterName, minorVersion, nodeMachineType, region, amiId string, subnetAvZones []string) *eksctlapi.ClusterConfig {
+	clusterCfg := eksctlapi.NewClusterConfig()
+
+	clusterCfg.Metadata.Name = clusterName
+	clusterCfg.Metadata.Region = region
+	clusterCfg.Metadata.Version = minorVersion
+	clusterCfg.KubernetesNetworkConfig.ServiceIPv4CIDR = DefaultKubernetesSvcCIDR
+	clusterCfg.Status = &eksctlapi.ClusterStatus{}
+
+	ng := clusterCfg.NewNodeGroup()
+	ng.Name = DefaultNodeGroupName
+	ng.ContainerRuntime = aws.String(eksctlapi.ContainerRuntimeContainerD)
+	ng.AMIFamily = eksctlapi.DefaultNodeImageFamily
+	ng.AMI = amiId
+	ng.InstanceType = nodeMachineType
+	ng.AvailabilityZones = subnetAvZones
+	ng.ScalingConfig = &eksctlapi.ScalingConfig{
+		DesiredCapacity: aws.Int(1),
+		MinSize:         aws.Int(1),
+		MaxSize:         aws.Int(1),
+	}
+
+	nodeKeyName := os.Getenv(envKeyNodeSSHKeyName)
+	if nodeKeyName != "" {
+		ng.SSH.Allow = aws.Bool(true)
+		ng.SSH.PublicKeyName = aws.String(nodeKeyName)
+	}
+
+	return clusterCfg
+}
+
+func waitForClusterActive(ctx context.Context, eksClient *eks.Client, clusterName string) (*types.Cluster, error) {
+	childCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	ticker := time.NewTicker(10 * time.Second)
```

Comment:
Is the 10 second ticker too long? Or should we make the wait time and wait tick configurable?

Reply:
Creating a cluster can take up to 10 minutes. So I assume 10 seconds is not too long.